data/collector.go

```diff
@@ -285,10 +338,60 @@ func (c *collector) writeCaptureResults() {
 				c.logger.Error(errors.Wrap(err, fmt.Sprintf("failed to write to collector %s", c.target.Path())).Error())
 				return
 			}
+
+			c.maybeWriteToMongo(msg)
 		}
 	}
 }
 
+// TabularData is a denormalized sensor reading.
+type TabularData struct {
+	TimeRequested time.Time `bson:"time_requested"`
+	TimeReceived  time.Time `bson:"time_received"`
+	ComponentName string    `bson:"component_name"`
+	ComponentType string    `bson:"component_type"`
+	MethodName    string    `bson:"method_name"`
+	Data          bson.M    `bson:"data"`
+}
+
+func (c *collector) maybeWriteToMongo(msg *v1.SensorData) {
```

Comment:
add a doc saying if anything fails here, we (maybe) log but always return - the action is completely optional

Reply:
done